src/librustdoc/clean/blanket_impl.rs

```diff
@@ -18,109 +19,102 @@ impl<'a, 'tcx> BlanketImplFinder<'a, 'tcx> {
 
         trace!("get_blanket_impls({:?})", ty);
         let mut impls = Vec::new();
-        self.cx.with_all_traits(|cx, all_traits| {
-            for &trait_def_id in all_traits {
+        self.cx.with_all_trait_impls(|cx, all_trait_impls| {
+            'blanket_impls: for &impl_def_id in all_trait_impls {
+                let trait_def_id = cx.tcx.trait_id_of_impl(impl_def_id).unwrap();
```

Comment:
I think the perf regression is because we're now processing the same trait multiple times, once for each impl. Can we have a function-local cache of traits to avoid processing them more than once?

Reply:
Ah no I'm wrong; we still used to go through each impl id on line 29 below.

Reply:
I've removed this change, it's not necessary for removing `EarlyDocLinkResolver::all_traits`.